internal/apiserver/pgadminservice/pgadminimpl.go

```diff
@@ -61,14 +61,60 @@ func CreatePgAdmin(request *msgs.CreatePgAdminRequest, ns, pgouser string) msgs.
 			return resp
 		}
 
+		// set the default pgadmin storage value (i.e. the primary storage value)
+		if cluster.Spec.PGAdminStorage == (crv1.PgStorageSpec{}) {
+			cluster.Spec.PGAdminStorage = cluster.Spec.PrimaryStorage
+			log.Info("pgAdmin storage not set, using primary storage configuration")
+		}
+
+		// if a value for pgAdmin storage config is provided with the request, validate it here
+		// if it is not valid, return now
+		if request.StorageConfig != "" && !apiserver.IsValidStorageName(request.StorageConfig) {
+			resp.Status.Code = msgs.Error
+			resp.Status.Msg = fmt.Sprintf("%q storage config provided invalid", request.StorageConfig)
+			return resp
+		}
+
+		// validate the pgAdmin storage config value in the main configuration if the request did not provide one
+		// if it is set but not valid, return now
+		if request.StorageConfig == "" && !apiserver.IsValidStorageName(apiserver.Pgo.PGAdminStorage) {
+			resp.Status.Code = msgs.Error
+			resp.Status.Msg = fmt.Sprintf("%q configured storage config invalid", apiserver.Pgo.PGAdminStorage)
+			return resp
+		}
+
+		// Extract parameters for the optional pgAdmin storage. server configuration and
+		// request parameters are all optional.
+		// Now, set the main configured value, if provided
+		if apiserver.Pgo.PGAdminStorage != "" {
+			cluster.Spec.PGAdminStorage, _ = apiserver.Pgo.GetStorageSpec(apiserver.Pgo.PGAdminStorage)
+		}
```

Comment:
We can just set this by default. If the user picked a one off, it's handled in the statement below this.

```suggestion
		cluster.Spec.PGAdminStorage, _ = apiserver.Pgo.GetStorageSpec(apiserver.Pgo.PGAdminStorage)
```

Reply:
sure thing, updated.